ABCDoubleE.Repositories/PreferenceRepository.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.EntityFrameworkCore;
+using ABCDoubleE.Models;
+using ABCDoubleE.Data;
+
+public class PreferenceRepository : IPreferenceRepository
+{
+    private readonly ABCDoubleEContext _context;
+
+    public PreferenceRepository(ABCDoubleEContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<Preference?> GetPreferenceByUserIdAsync(int userId)
+    {
+        return await _context.Preferences
+            .Include(p => p.favBooks)
+            .FirstOrDefaultAsync(p => p.userId == userId);
+    }
+
+    public async Task<Preference?> GetPreferenceByIdAsync(int preferenceId)
+    {
+        return await _context.Preferences
+            .Include(p => p.favBooks)
+            .FirstOrDefaultAsync(p => p.preferenceId == preferenceId);
+    }
+
+    public async Task AddPreferenceAsync(Preference preference)
+    {
+        await _context.Preferences.AddAsync(preference);
+        await _context.SaveChangesAsync();
+    }
+
+
+    public async Task UpdatePreferenceAsync(Preference preference)
+    {
+        _context.Preferences.Update(preference);
```

Comment:
should there be an await here?

Reply:
The Update() method in Entity Framework is a synchronous method. It doesn't perform any database operations by itself. Instead, it marks the entity as modified